### PR TITLE
Cloudnative smf and amf integration test fixes

### PR DESCRIPTION
**Pull request by vthiruveedula**

### CODE REVIEW

File: context/db.go [hunk 1]
```diff
@@ -332,10 +332,13 @@ func GetSMContextBySEIDInDB(seidUint uint64) (smContext *SMContext) {
 	filter["seid"] = seid
 
 	result := MongoDBLibrary.RestfulAPIGetOne(SeidSmContextCol, filter)
-	ref := result["ref"].(string)
-	fmt.Println("StoreSeidContextInDB, result string : ", ref)
-
-	return GetSMContext(ref)
+	if result != nil {
+		ref := result["ref"].(string)
+		fmt.Println("StoreSeidContextInDB, result string : ", ref)
```
Comment (thakurajayL) [inline context/db.go:337]: No need to have this Println.

File: context/db.go [hunk 2]
```diff
@@ -347,10 +350,12 @@ func DeleteSmContextInDBBySEID(seidUint uint64) {
...
 	result := MongoDBLibrary.RestfulAPIGetOne(SeidSmContextCol, filter)
-	ref := result["ref"].(string)
+	if result != nil {
+		ref := result["ref"].(string)
 
-	MongoDBLibrary.RestfulAPIDeleteOne(SeidSmContextCol, filter)
-	DeleteSmContextInDBByRef(ref)
+		MongoDBLibrary.RestfulAPIDeleteOne(SeidSmContextCol, filter)
+		DeleteSmContextInDBByRef(ref)
+	}
```
Comment (thakurajayL) [inline context/db.go:358]: Error case you should have log.

File: pfcp/adapter/adapter.go [hunk 3]
```diff
@@ -39,7 +39,7 @@ func InsertPfcpTxn(seqNo uint32, upNodeID *pfcpType.NodeID) {
 	PfcpTxns[seqNo] = upNodeID
 }
 
-func HandleAdapterPfcpRsp(pfcpMsg pfcp.Message) error {
+func HandleAdapterPfcpRsp(pfcpMsg pfcp.Message, evtData *pfcpUdp.PfcpEventData) error {
```
Comment (thakurajayL) [inline pfcp/adapter/adapter.go:42]: Comment at start..why we don't need lock on upf in this function. Assumption is that this is synchronous call and called is already holding the upf lock ?

File: pfcp/adapter/adapter.go [hunk 4]
```diff
@@ -197,14 +192,14 @@ func HandlePfcpSessionEstablishmentResponse(msg *pfcpUdp.Message) {
...
 	}
 	smContext := smf_context.GetSMContextBySEID(SEID)
-	logger.PfcpLog.Infoln("In HandlePfcpSessionEstablishmentResponse SEID %v", SEID)
+	logger.PfcpLog.Infoln("In HandlePfcpSessionEstablishmentResponse SEID ", SEID)
 	logger.PfcpLog.Infoln("In HandlePfcpSessionEstablishmentResponse smContext %v", smContext)
 	if rsp.UPFSEID != nil {
 		NodeIDtoIP := rsp.NodeID.ResolveNodeIdToIp().String()
 		pfcpSessionCtx := smContext.PFCPContext[NodeIDtoIP]
 		pfcpSessionCtx.RemoteSEID = rsp.UPFSEID.Seid
+		smContext.SubPfcpLog.Infof("in adt HandlePfcpSessionEstablishmentResponse rsp.UPFSEID.Seid [%v] ", rsp.UPFSEID.Seid)
```
Comment (thakurajayL) [inline pfcp/adapter/adapter.go:201]: what is "adt" in this log ?

**Review by thakurajayL — CHANGES_REQUESTED** [review]

**Review by UpadhyayaAnkur — DISMISSED** [review]

File: context/db.go [hunk 5]
```diff
@@ -332,10 +332,13 @@ func GetSMContextBySEIDInDB(seidUint uint64) (smContext *SMContext) {
 	filter["seid"] = seid
 
 	result := MongoDBLibrary.RestfulAPIGetOne(SeidSmContextCol, filter)
-	ref := result["ref"].(string)
-	fmt.Println("StoreSeidContextInDB, result string : ", ref)
-
-	return GetSMContext(ref)
+	if result != nil {
```
Comment (ajaythakurintel) [inline context/db.go:335]: Can you take care of the warning above ?

File: context/db.go [hunk 6]
```diff
@@ -332,10 +332,13 @@ func GetSMContextBySEIDInDB(seidUint uint64) (smContext *SMContext) {
...
+		ref := result["ref"].(string)
+		return GetSMContext(ref)
+	} else {
+		fmt.Println("DB entry doesn't exist with seid: ", seid)
```
Comment (ajaythakurintel) [inline context/db.go:339]: remove print.

File: context/db.go [hunk 7]
```diff
@@ -347,10 +350,14 @@ func DeleteSmContextInDBBySEID(seidUint uint64) {
...
-	ref := result["ref"].(string)
+	if result != nil {
+		ref := result["ref"].(string)
 
-	MongoDBLibrary.RestfulAPIDeleteOne(SeidSmContextCol, filter)
-	DeleteSmContextInDBByRef(ref)
+		MongoDBLibrary.RestfulAPIDeleteOne(SeidSmContextCol, filter)
+		DeleteSmContextInDBByRef(ref)
+	} else {
+		fmt.Println("DB entry doesn't exist with seid: ", seid)
```
Comment (ajaythakurintel) [inline context/db.go:359]: remove print.

File: pfcp/adapter/adapter.go [hunk 8]
```diff
@@ -197,14 +196,14 @@ func HandlePfcpSessionEstablishmentResponse(msg *pfcpUdp.Message) {
...
-	logger.PfcpLog.Infoln("In HandlePfcpSessionEstablishmentResponse SEID %v", SEID)
+	logger.PfcpLog.Infoln("In HandlePfcpSessionEstablishmentResponse SEID ", SEID)
 	logger.PfcpLog.Infoln("In HandlePfcpSessionEstablishmentResponse smContext %v", smContext)
 	if rsp.UPFSEID != nil {
 		NodeIDtoIP := rsp.NodeID.ResolveNodeIdToIp().String()
 		pfcpSessionCtx := smContext.PFCPContext[NodeIDtoIP]
 		pfcpSessionCtx.RemoteSEID = rsp.UPFSEID.Seid
+		smContext.SubPfcpLog.Infof("in HandlePfcpSessionEstablishmentResponse rsp.UPFSEID.Seid [%v] ", rsp.UPFSEID.Seid)
 	}
-	smContext.SubPfcpLog.Infof("in adt HandlePfcpSessionEstablishmentResponse rsp.UPFSEID.Seid [%v] ", rsp.UPFSEID.Seid)
```
Comment (ajaythakurintel) [inline pfcp/adapter/adapter.go:207]: Fix the warning coming out of GitHub Actions.

**Review by ajaythakurintel — CHANGES_REQUESTED** [review]

**Review by ajaythakurintel — APPROVED** [review]

**Review by thakurajayL — APPROVED** [review]

**Pull request merged**